scipy/optimize/tests/test_optimize.py

```diff
@@ -848,6 +848,52 @@ def cons(x):
                                 constraints={'type': 'ineq', 'fun': cons})
         assert_allclose(res.x, np.array([0., 2, 5, 8])/3, atol=1e-12)
 
+    @pytest.mark.parametrize('method', ['Nelder-Mead', 'Powell', 'CG', 'BFGS', 'Newton-CG', 'L-BFGS-B',
+                                        #  'TNC', 'SLSQP',
```

Comment:
Do TNC and SLSQP not pass this test?

Reply:
They don't pass the test. I wonder if `maxiter` in those methods should be `maxiter - 1`.